web/src/main/java/org/wso2/testgrid/web/api/APIUtil.java

```diff
@@ -72,16 +74,18 @@ static Product getProductBean(org.wso2.testgrid.common.Product product) {
      * {@link DeploymentPattern} instance.
      *
      * @param deploymentPattern {@link org.wso2.testgrid.common.DeploymentPattern} instance to be converted
-     * @param status Status of the last test execution
+     * @param status            Status of the last test execution
      * @return {@link DeploymentPattern} instance with necessary information
      */
     static DeploymentPattern getDeploymentPatternBean(org.wso2.testgrid.common.DeploymentPattern deploymentPattern,
                                                       String status) {
         DeploymentPattern deploymentPattern1 = new DeploymentPattern();
-        deploymentPattern1.setId(deploymentPattern.getId());
-        deploymentPattern1.setName(deploymentPattern.getName());
-        deploymentPattern1.setProductId(deploymentPattern.getProduct().getId());
-        deploymentPattern1.setTestStatus(status);
+        if (deploymentPattern != null) {
```

Comment:
Shall we change the variable names `deploymentPattern` and `deploymentPattern1` to something more meaningful since it is hard to distinguish which one is which at the moment